### ISSUE TITLE
Outputs generated with `sample_at_first` are noticeably lower in quality compared to direct inference

### ISSUE THREAD

**Issue by kingnobro** [kohya-ss/sd-scripts#2181]
I’m training a Flux LoRA and ran into a quality issue.

I set `sample_at_first = true`, so the script generates sample prompts at the beginning (before training happens). However, I noticed a clear downgrade in output quality compared to direct inference.

For testing, I used this prompt (parameters are at the end of the prompt):

```
masterpiece, 4K, sharp and detailed, high resolution, best quality, an astronaut riding a camel, sun is shining, trending on artstation --w 512 --h 512 --d 10 --l 3.5 --s 30
```

I generated three images (with the same prompt, width, height, seed, cfg, steps):

* one using `flux_train_utils.py` (this is automatically called at the beginning of the training)
* one using `flux_minimal_inference.py`
* one using Diffusers

The results differ noticeably, and the training utility output looks worse. I’ve been stuck on this for hours. Does anyone know why this happens?

<img width="1266" height="463" alt="Image" src="https://github.com/user-attachments/assets/f51eabca-9f79-4313-bbf9-646ee30723a1" />

This is the shell for training:
```
accelerate launch --config_file config/train.yaml --num_cpu_threads_per_process 1 flux_train_network.py \
  --pretrained_model_name_or_path="models/flux1-dev.safetensors" \
  --clip_l="models/clip_l.safetensors" \
  --t5xxl="models/t5xxl_fp16.safetensors" \
  --ae="models/ae.safetensors" \
  --dataset_config="config/dataset.toml" \
  --output_dir="output" \
  --output_name="lora" \
  --save_model_as=safetensors \
  --network_module=networks.lora_flux \
  --network_dim=32 \
  --network_alpha=1 \
  --learning_rate=1e-4 \
  --optimizer_type="AdamW8bit" \
  --lr_scheduler="constant" \
  --max_train_epochs=10 \
  --save_every_n_epochs=1 \
  --mixed_precision="bf16" \
  --gradient_checkpointing \
  --guidance_scale=1.0 \
  --timestep_sampling="flux_shift" \
  --model_prediction_type="raw" \
  --log_with=wandb \
  --wandb_run_name="test" \
  --sample_at_first \
  --sample_every_n_steps=500 \
  --sample_prompts="sample_prompts.txt" \
  --sample_sampler="euler" \
```

This is the shell for direct inference:
```shell
python flux_minimal_inference.py \
  --ckpt_path="models/flux1-dev.safetensors" \
  --model_type="flux" \
  --clip_l="models/clip_l.safetensors" \
  --t5xxl="models/t5xxl_fp16.safetensors" \
  --ae="models/ae.safetensors" \
  --prompt="masterpiece, 4K, sharp and detailed, high resolution, best quality, an astronaut riding a camel, sun is shining, trending on artstation" \
  --output_dir="output/inference" \
  --dtype="bfloat16" \
  --seed=10 \
  --steps=30 \
  --guidance=3.5 \
  --width=512 \
  --height=512 \
  --offload
```

Diffusers:
```python
import torch
from diffusers import FluxPipeline

pipe = FluxPipeline.from_pretrained("black-forest-labs/FLUX.1-dev", torch_dtype=torch.bfloat16)
pipe.enable_model_cpu_offload() #save some VRAM by offloading the model to CPU. Remove this if you have enough GPU power

prompt = "masterpiece, 4K, sharp and detailed, high resolution, best quality, an astronaut riding a camel, sun is shining, trending on artstation"
image = pipe(
    prompt,
    height=512,
    width=512,
    guidance_scale=3.5,
    num_inference_steps=30,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(10),
).images[0]
image.save("flux-dev.png")
```

**Comment by Ice-YY** [comment]
I think you need --g instead of --l

**Comment by kohya-ss** [comment]
Sorry, this is a breaking change in the latest code. Please see the documentation here for more details.

https://github.com/kohya-ss/sd-scripts/tree/sd3?tab=readme-ov-file#sample-image-generation-during-training

**Comment by kingnobro** [comment]
Thank you! The problem is solved.
